src/middlewares/Authenticate.js

```diff
@@ -0,0 +1,55 @@
+import { verifyToken, errorResponse } from '../utils';
+import models from '../models';
+
+const { Profiles, Sequelize } = models;
+/**
+ * @class Authenticate
+ * @description authenticate tokens and roles
+ * @exports Authenticate
+ */
+class Authenticate {
+  /**
+   * Verify if token is valid
+   * @param  {object} req - The user request object
+   * @param  {object} res - The user res response object
+   * @param  {function} next - The next() Function
+   * @returns {String} req.userId - The user id
+   */
+  static async verifyToken(req, res, next) {
+    try {
+      const { headers: { authorization } } = req;
+      const token = authorization.split(' ')[1];
+      const decoded = await verifyToken(token);
+      req.user = decoded;
```

Comment:
You need to improve this part of the code to check when decoded is false. The error message in the `catch` block is what should be returned when we cannot verify the token. E.g.
```
const decoded = await verifyToken(token);
if (!decoded) {
  return errorResponse(res, 401, 'Access denied. We could not verify your identity');
}
req.user = decoded
return next()
```

The catch block is normally used to handle server error.